pkg/microservice/aslan/core/system/service/registry.go

```diff
@@ -199,44 +203,186 @@ func ListAllRepos(log *zap.SugaredLogger) ([]*RepoInfo, error) {
-		Endpoint: registry.Endpoint{
-			Addr:      registryInfo.RegAddr,
-			Ak:        registryInfo.AccessKey,
-			Sk:        registryInfo.SecretKey,
-			Namespace: registryInfo.Namespace,
-			Region:    registryInfo.Region,
-		},
-		Repos: names,
+		Endpoint: endPoint,
+		Repos:    names,
 	}, logger)
 
 	images := make([]*RepoImgResp, 0)
 	if err == nil {
 		for _, repo := range repos.Repos {
-			for _, tag := range repo.Tags {
-				img := &RepoImgResp{
-					Host:  util.TrimURLScheme(registryInfo.RegAddr),
-					Owner: repo.Namespace,
-					Name:  repo.Name,
-					Tag:   tag,
+			// find all tags from image tags db
+			imageTags, err := commonrepo.NewImageTagsCollColl().Find(&commonrepo.ImageTagsFindOption{
+				RegistryID:  registryInfo.ID.Hex(),
+				RegProvider: registryInfo.RegProvider,
+				ImageName:   repo.Name,
+				Namespace:   registryInfo.Namespace,
+			})
+			dbTags := make(map[string]*commonmodels.ImageTag, 0)
+			if err != nil {
+				logger.Errorf("find image[%s] tags from db failed, the error is: %v", repo.Name, err)
+			} else {
+				for _, imageTag := range imageTags.ImageTags {
+					dbTags[imageTag.TagName] = imageTag
 				}
-				images = append(images, img)
 			}
+			logger.Infof("get image[%s] %d tags from db, get image[%s] %d tags from registry", repo.Name, len(dbTags), repo.Name, len(repo.Tags))
+
+			tags := imageProcessor(repo, registryInfo, dbTags, regService, logger)
+			// update imageTags db, we don't care about the result
+			go insertNewTags2DB(tags, registryInfo, logger)
+
+			images = append(images, tags...)
 		}
+		// sort tags by created time
+		sort.Slice(images, func(i, j int) bool {
+			return images[i].Created > images[j].Created
+		})
 	} else {
 		err = e.ErrListImages.AddErr(err)
 	}
 
 	return images, err
 }
 
+func imageProcessor(repo *registry.Repo, registryInfo *commonmodels.RegistryNamespace, dbTags map[string]*models.ImageTag, regService registry.Service, logger *zap.SugaredLogger) []*RepoImgResp {
+	resultChan := make(chan *RepoImgResp, 10)
+	defer close(resultChan)
+	jobChan := make(chan *imageReqJob, len(repo.Tags))
+	images := make([]*RepoImgResp, 0)
+
+	jobCount := 0
+	for _, tag := range repo.Tags {
+		if t, ok := dbTags[tag]; ok {
+			resp := &RepoImgResp{
+				Host:    util.TrimURLScheme(registryInfo.RegAddr),
+				Owner:   repo.Namespace,
+				Name:    repo.Name,
+				Tag:     t.TagName,
+				Created: t.Created,
+				Digest:  t.Digest,
+			}
+			images = append(images, resp)
+			continue
+		}
+
+		jobCount++
+		jobChan <- &imageReqJob{
+			regService: regService,
+			endPoint: registry.Endpoint{
+				Addr:      registryInfo.RegAddr,
+				Ak:        registryInfo.AccessKey,
+				Sk:        registryInfo.SecretKey,
+				Namespace: registryInfo.Namespace,
+				Region:    registryInfo.Region,
+			},
+			registryInfo: registryInfo,
+			repoImages:   repo,
+			tag:          tag,
+		}
+	}
+
+	// create gcount goroutines to get tags from registry
```

Comment:
There is still has room to impove go and chan, use waitGroup and forrange could be better